front/src/modules/simulationResult/components/SpeedSpaceChart/useSpeedSpaceChart.ts

```diff
@@ -0,0 +1,216 @@
+            powerRestrictionsRanges,
+            formattedPathProps.electrifications,
+            rollingStock.effort_curves.modes
+          );
+
+          setFormattedPowerRestrictions(powerRestrictionsWithHandled);
+        }
+      }
+    };
+
+    getPathProperties();
+  }, [pathfindingResult, simulation, infraId, trainScheduleResult, rollingStock]);
+
+  // setup chart synchronizer
+  useEffect(() => {
+    if (simulation?.status === 'success' && trainScheduleResult && rollingStock && departureTime) {
+      const {
+        baseHeadPositions,
+        baseTailPositions,
+        baseSpeeds,
+        marginSpeeds,
+        ecoHeadPosition,
+        ecoTailPosition,
+        ecoSpeeds,
+      } = simulation.base.positions.reduce(
+        (results, position, index) => {
+          const positionInMeters = mmToM(position);
+
+          // TODO GET v2 : probably remove this conversion as trains will travel on several days
+          // The chart time axis is set by d3 function *sec2d3datetime* which start the chart at 01/01/1900 00:00:00
+          // As javascript new Date() util takes count of the minutes lost since 1/1/1900 (9min and 21s), we have
+          // to use sec2d3datetime here as well to set the times on the chart
+          const time = sec2d3datetime(
+            isoDateWithTimezoneToSec(departureTime) + simulation.base.times[index] / 1000
+          );
+
+          if (!time) {
+            return results;
+          }
+
+          results.baseHeadPositions.push({ position: positionInMeters, time });
+          results.baseTailPositions.push({
+            position: positionInMeters - rollingStock.length,
+            time,
+          });
+          results.baseSpeeds.push({
+            position: positionInMeters,
+            time,
+            speed: simulation.base.speeds[index],
+          });
+          results.marginSpeeds.push({
+            position: positionInMeters,
+            time,
+            speed: simulation.final_output.speeds[index],
+          });
+          results.ecoHeadPosition.push({ position, time });
+          results.ecoTailPosition.push({
+            position: positionInMeters - rollingStock.length,
+            time,
+          });
+          results.ecoSpeeds.push({
+            position: positionInMeters,
+            time,
+            speed: simulation.final_output.speeds[index],
+          });
+          return results;
+        },
+        {
+          baseHeadPositions: [] as { time: Date; position: number }[],
+          baseTailPositions: [] as { time: Date; position: number }[],
+          baseSpeeds: [] as { time: Date; position: number; speed: number }[],
+          marginSpeeds: [] as { time: Date; position: number; speed: number }[],
+          ecoHeadPosition: [] as { time: Date; position: number }[],
+          ecoTailPosition: [] as { time: Date; position: number }[],
+          ecoSpeeds: [] as { time: Date; position: number; speed: number }[],
+        }
+      );
```

Comment:
I would split this in another function processData

Reply:
Done :+1: